src/LicenseChecker.php

```diff
@@ -146,20 +161,41 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                 }
 
                 $violation = true;
-                $style->error(
-                    \sprintf(
-                        'Dependency "%s" has license "%s" which is not in the allowed list.',
-                        $package->name->toString(),
-                        $license,
-                    ),
-                );
+                $dispatcher->dispatch(new PackageWithViolatingLicense($package, $license));
             }
         }
 
-        if (!$violation) {
-            $style->success('All dependencies have allowed licenses.');
+        if ($violation) {
+            $dispatcher->dispatch(new OutcomeFailure());
+
+            return self::FAILURE;
         }
 
-        return $violation ? self::FAILURE : self::SUCCESS;
+        $dispatcher->dispatch(new OutcomeSuccess());
+
+        return self::SUCCESS;
+    }
+
+    private function createDispatcher(InputInterface $input, OutputInterface $output): Dispatcher
+    {
+        $dispatcher = new Dispatcher();
+
+        $format = $input->getOption('format');
+        \assert(\is_string($format));
+
+        $outputSubscriber = match ($format) {
+            'human' => new HumanReadableSubscriber($input, $output),
```

Comment:
It's all beautiful but we could make it even more fantastic by merging it with Display and have single Subscriber handling the output formatting.